### ISSUE TITLE
dynamic content with shortcodes

### ISSUE THREAD

**Issue by buffdownunder** [pojome/elementor#5679]
There is a thread that inspired me but it is closed
https://github.com/pojome/elementor/issues/4181

**What problem is your feature request going to solve? Please describe.**

As soon as you go beyond a website with a few landing pages you have the problem that you don't want to open up and manually edit all the widgets just to change one tiny setting.

I would love to define a Products template and then alter the content by adding parameters to the shortcode. This way I could reuse styles while being able to change the content displayed.

**Describe the solution you'd like**

Something similar to the shortcode options of CVPro.
Something like this
[elementor-template id="7972" taxonomy="product_cat" terms="high-uv-buff" taxonomy2="product_tag" terms2="blue"]
So basically take the template 7972 and display Woocommerce products that fit the category High UV Buff® and the tag "blue"
Similar stuff I would like to do with the Posts Widget
Is this already possible or is this a feature request?

**Describe alternatives you've considered**

Editing CSS of the widget (not good), playing with Global Widget (Can't change content). One other alternative would be to provide settings templates for widgets. This could be achieved with a search & replace function that scans all content and searches for patterns and then changes the setting according to matching patterns. I have mentioned this somewhere else.

**Additional context**

**Comment by bainternet** [comment]
Currently, this feature is not supported, but we will look into it in the future if we see a large demand for it.

Thanks for the feedback.

**Comment by mikesale** [comment]
This would be great, especially given that the only way to have a section global is via shortcodes!

**Comment by uuykay** [comment]
+1 for this

**Comment by mice777** [comment]
All that would be needed is support for Dynamic parser of shortcode parameters, similar to existing `Request Parameter` dynamic item. 
Then whatever I'd pass as parameter for template's shortcode would be expanded.

I think it would look like this:
![image](https://user-images.githubusercontent.com/3145438/72608708-95516700-3923-11ea-832a-103ffeddbde8.png)

**Comment by mice777** [comment]
I actually implemented this with little piece of code.

Theme's `functions.php`:
```
/* Call another shortcode with provided parameters. */
add_shortcode('lcg', function ($atts = [], $content = null){
	global $lcg_params;
	$save_global = $lcg_params; // save globals for possible recursion
	$lcg_params = $atts;
	$content = do_shortcode($content, false);
	$lcg_params = $save_global;
    return $content;
});

/* Fetch parameter from template or page. */
add_shortcode('lcg-param', function ($atts = [], $content = null){
	global $lcg_params;
	return $lcg_params[$atts['p']];
});
```

Then enclose elementor's template shortcode in `lcg` shortcode:
`[lcg title1="Hello"][elementor-template id='614'][/lcg]`
there is start and end tag for `lcg` shortcode, with all desired parameters passed in. Inside is original template's shortcode.

In the template, you can pull actual parameter with another shortcode:
`[lcg-param p='title1']`
In this case, it will be replaced by "Hello" passed in template invocation.

And that's it, we have function-like parametrized template instances!

Hope one day Elementor adds this inside of its UI without need of this trick.

**Comment by johnrisby** [comment]
Thanks mice777 that's exactly what I was looking for! Shame it's not an in-built feature in Elementor though so we could edit the variables in fields (or live on the page) rather than in parameters. Thanks again.

**Comment by veryaca** [comment]
This works fine on the front end, but in the backend, I see the shortcode showing twice (in Elementor), and it returns an error when loading the page without Elementor (directly in WP).

**Comment by johnrisby** [comment]
Hope it's ok to mention this here. I've been looking for a solution for Divi which I've recently started using as well as Elementor (much improved since last year btw). I was in need of this feature because Divi also lacks it. As I use this on Elementor I wondered if it would work on Divi and sure enough, it just works. No changes needed. So just popping this comment here to help people find this solution when searching on Google. Thanks (again) @mice777 for the function!
